src/actions/controller.ts

```diff
@@ -29,16 +28,63 @@ export class ActionsController {
     return ActionsController.actions.find(action => action.id === id);
   }
 
-  public static async performActionsFromParams() {
+  public static async performActionsFromURL() {
     const actionsParam = await getParameter("actions");
-    const actionsIds = uniq(compact(split(actionsParam, ",")));
-    getLogger().trace(`configuration - actionsIds= ${actionsIds}`, { method: "performActionsFromParams" });
+    if (actionsParam === undefined) {
+      return;
+    }
+    const decodedActionsParam = decodeURI(actionsParam);
+    getLogger().trace(`decodedActionsParam= ${decodedActionsParam}`, { method: "performActionsFromURL" });
+    const mode = ActionsController.detectActionMode(decodedActionsParam);
+    switch (mode) {
+      case "IDs": {
+        const actionsIds = uniq(compact(split(decodedActionsParam, ",")));
+        ActionsController.performActionsIds(actionsIds);
+        break;
+      }
+      case "Inlined": {
+        ActionsController.perfomInlinedActions(decodedActionsParam.trim());
+        break;
+      }
+    }
+  }
+
+  private static detectActionMode(decodedActionsParam:string): "IDs" | "Inlined" {
+    try {
+      if (Array.isArray(JSON.parse(decodedActionsParam))) {
+        // actionsInlinedMode
+        // actions=[{"id":"openSettings","actionType":"COMMAND","name":"workbench.action.openSettings"},{"actionType":"FILE","uri":"https://www.google.com/"}]
+        return "Inlined";
+      }
+    }
+    catch (e) {
+      // actionsIDsMode
+      //actions=openSettings,openGoogle
+    }
+    return "IDs";
+  }
+
+  private static performActionsIds(actionsIds: string[]) {
```

Comment:
what do you think about naming this function: performActions**By**Ids

So we have:
- perform**Inlined**Actions
- performActions**ByIds**

Which makes more sense from English grammar perspective.